crates/api/src/gossip.rs

```diff
@@ -0,0 +1,35 @@
+//! Gossip related types.
+
+use crate::peer_store::DynPeerStore;
+use crate::transport::{DynTransport, DynTxModuleHandler};
+use crate::{builder, config, BoxFut, DynOpStore, K2Result, SpaceId};
+use std::sync::Arc;
+
+/// Represents the ability to sync DHT data with other agents through background communication.
+pub trait Gossip: 'static + Send + Sync + std::fmt::Debug {
+    /// Get the tx module handler for this gossip instance.
+    fn tx_module_handler(&self) -> DynTxModuleHandler;
```

Comment:
This I'm not quite sure about. I want to be able to have the gossip implementation also act as a module message receiver. That is think is reasonable but I'm wondering if there was a different way of achieving that already envisioned

Reply:
I was thinking either the factory::create or the constructor for the internal gossip module instance type could do the registering of the module callback (since they are passed the transport instance) so that wouldn't have to be exposed outside of the module.

Reply:
That makes a lot more sense, done!